### PR TITLE
Upgrade pandoc to 2.14 and fix plugin breaks

### PR DESCRIPTION
**Pull request by vincerubinetti**
supersedes #425

### CODE REVIEW

File: build/plugins/anchors.html
Comment (dhimmel) [inline build/plugins/anchors.html:14]: Is `div[id^="fig:"]`the workaround for double table ids (on the div and table) mentioned at https://github.com/manubot/rootstock/pull/425#issuecomment-853395226? It works by only matching div ids that start with `tbl:` as opposed to all elements?

Just to confirm, it is okay for us to proceed despite the duplicate ids, because this PR handles it?

File: build/themes/default.html
Comment (dhimmel) [inline build/themes/default.html:504]: I think I'd be in favor of a slightly smaller margin between references than:

image

I think we do want some, but since reference sections can get quite long, perhaps we could get by with slightly less? Maybe 15 (or 50% to 75% of the current spacing)?

**Review by dhimmel — COMMENTED** [review]
Reviewing https://ci.appveyor.com/api/buildjobs/992931201tet4bko/artifacts/manuscript-1.0.244-b5233d9.html and everything looks good (one small comment about references spacing).

Thanks a lot!

File: build/plugins/anchors.html
Comment (dhimmel) [reply build/plugins/anchors.html:14]: Opened https://github.com/tomduck/pandoc-tablenos/issues/31

Comment (vincerubinetti) [reply build/plugins/anchors.html:14]: > Is div[id^="fig:"]the workaround for double table ids (on the div and table) mentioned at #425 (comment)? It works by only matching div ids that start with tbl: as opposed to all elements?

That's correct. I've essentially made the queries more specific. As long as pandoc/table-nos is consistent and never outputs a table without the div wrapper, this should be fine.

Looked at the issue, 👍 FWIW id's being unique isn't just a suggestion, it's a requirement of the spec and can cause unexpected behavior. It shouldn't cause anything to crash though.

File: build/themes/default.html
Comment (vincerubinetti) [reply build/themes/default.html]: Reduced to 15px. I wouldn't recommend going lower than that, as it becomes difficult to distinguish for sighted and hard-of-seeing people alike.

File: build/plugins/anchors.html
Comment (dhimmel) [reply build/plugins/anchors.html:14]: Got it.

> As long as pandoc/table-nos is consistent and never outputs a table without the div wrapper, this should be fine.

Based on the early discussion in the issue, it seems like the most likely fix is that pandoc-tablenos stops adding the div wrapper for tables since the id is now set in `<figure>`.

So if we upgrade pandoc-tablenos, we might have to tweak this, but that's okay. Let's wait to see what solution pandoc-tablenos adopts and merge this now.

File: build/themes/default.html
Comment (dhimmel) [reply build/themes/default.html]: Here's 15. Looks good:

image

File: build/plugins/anchors.html
Comment (vincerubinetti) [reply build/plugins/anchors.html:14]: Oh if that’s the case we’ll need to re-add the plug-in to wrap tables in divs for scrolling purposes 😞

**Review by dhimmel — APPROVED** [review]
LGTM. Will wait for @agitter  to review before merging

**Comment by dhimmel** [comment]
@agitter following up in case you'd like to review

File: USAGE.md
Comment (agitter) [inline USAGE.md:136]:
```suggestion
The citation key syntax is described in the [Pandoc manual](https://pandoc.org/MANUAL.html#citation-syntax):
```

Is this part of the manual more relevant now?

File: build/environment.yml
Comment (agitter) [inline build/environment.yml:31]: Do we want to get this from conda?  2.1.0 is now available through conda-forge: https://anaconda.org/conda-forge/panflute

**Review by agitter — APPROVED** [review]
Looks good to me.  I have two minor suggestions, both optional.

**Comment by agitter** [comment]
The builds are failing because adding panflute to the conda-installed packages revealed an actual conflict.  panflute 2.1 [supports](https://github.com/sergiocorreia/panflute#supported-pandoc-versions) pandoc versions 2.11.0.4—2.13.x.  When we install panflute from pip, it can't check the pandoc version.  The conda-forge recipe does check the pandoc version.

Our environment appears to work based on the HTML artifacts above when we ran with panflute 2.1.0 and pandoc 2.14.0.1.  Should we revert 9ae8d19a91801713c8a85b70176945024c7dc485 and proceed to merge?

**Comment by dhimmel** [comment]
Thanks all for the help! Especially @vincerubinetti for the frontend edits. The diff looks simple in the end, but this includes some major upgrades in our dependencies.

**Pull request merged**
